src/core/lib/resource_quota/memory_quota.h

```diff
@@ -295,7 +297,11 @@ class GrpcMemoryAllocatorImpl final : public EventEngineMemoryAllocatorImpl {
     // Add the released memory to our free bytes counter... if this increases
     // from  0 to non-zero, then we have more to do, otherwise, we're actually
     // done.
-    if (free_bytes_.fetch_add(n, std::memory_order_release) != 0) return;
+    size_t prev_free = free_bytes_.fetch_add(n, std::memory_order_release);
+    if (prev_free + n > 1024 * 1024) {
```

Comment:
Let's make a symbolic constant for 1024*1024... kMaxQuotaBufferSize?

Reply:
Done.